js/runtime.ts

```diff
@@ -147,11 +152,31 @@ export function resolveModule(
 ): null | FileModule {
   util.log("resolveModule", { moduleSpecifier, containingFile });
   util.assert(moduleSpecifier != null && moduleSpecifier.length > 0);
-  // We ask golang to sourceCodeFetch. It will load the sourceCode and if
-  // there is any outputCode cached, it will return that as well.
-  const fetchResponse = os.codeFetch(moduleSpecifier, containingFile);
-  const { filename, sourceCode, outputCode } = fetchResponse;
-  if (sourceCode.length === 0) {
+  let filename: string, sourceCode: string, outputCode: string;
+  if (
+    moduleSpecifier.startsWith("/$asset$/") ||
+    containingFile.startsWith("/$asset$/")
+  ) {
+    // Assets are compiled into the runtime javascript bundle.
+    const assetName = moduleSpecifier.split("/").pop();
+    sourceCode = assetSourceCode[assetName];
+    filename = "/$asset$/" + assetName;
+  } else {
+    // We ask golang to sourceCodeFetch. It will load the sourceCode and if
+    // there is any outputCode cached, it will return that as well.
+    let fetchResponse;
+    try {
+      fetchResponse = os.codeFetch(moduleSpecifier, containingFile);
+    } catch (e) {
+      // TODO Only catch "no such file or directory" errors. Need error codes.
```

Comment:
Just having spent an hour debugging an error that was being swallowed here, I feel that this is not very helpful.

Reply:
I'm just porting the code over with as few changes as possible. I think we should keep it for now.

Reply:
Could you make it print the error message at least?

Reply:
I uncommented the `util.log` here - which is only displayed when the debug flag in util.ts is changed - but that should help debugging for now.